scripts/src/main/resources/scripts/command/docker

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+
+# autocompletion list
+if [ "${1}" = "shortlist" ]
+then
+  if [ -z "${2}" ]
+  then
+    echo "setup help"
+  fi
+  exit
+fi
+
+if [ -n "${DEVON_IDE_TRACE}" ]; then set -vx; fi
+# shellcheck source=scripts/functions
+source "$(dirname "${0}")"/../functions
+
+# Initiates docker setup
+function doSetup() {
+  if doIsBatch
+  then
+    doFail "Interactive installation is required, cannot proceed in batch mode. Please rerun without batch option."
+  fi
+  if doIsWindows
+  then 
+    wsl_distro_list=$(wsl -l 2>/dev/null)
+    if [[ $wsl_distro_list == *"no installed distributions"* ]];then
+      doEcho "No Linux distributions found in WSL or WSL is disabled. Please enable WSL and install any linux distribution to proceed with docker setup."
+      exit 0
+    fi
+    docker_win_version=$(docker version 2>/dev/null)
+    if [[ $docker_win_version == *"Client:"* ]];then
+      doEcho Docker already installed on Windows.
+      docker version
+      if [[ $docker_win_version != *"Server:"* ]];then
+      doEcho Please start Docker engine.
+      exit 0
+      fi
+      exit 0
+    else 
+      doEcho Docker not installed on Windows.
+    fi  
+    # Install Docker on WSL
+    wsl bash ./scripts/setup-docker
+  else
+    doEcho "Sorry, docker-installation support is not yet implemented for your OS. Please install manually or help devonfw-ide to support it for your OS by contributing a pull-request."
+    exit 0
+  fi
+}
+
+# Call Docker with specified arguments
+function doRun() {
+  doEcho "Running: docker ${*}"
```

Comment:
here we first should call `doSetup` so that if docker is not already installed, it should first be installed.

Reply:
> @isandesh1986 thanks for your improvements. This is now perfectly following the philosophy of devonfw-ide and all the other existing commandlets.
> Still I have left some last review comments. Also in some places indent is by 4 spaces and could be aligned. Will you be available to update PR once more? Otherwise let me know when I need to adapt the PR directly...

Hi @hohwille  - I will update the PR and revert. Thanks